bytecomp/translmod.ml

```diff
@@ -197,40 +204,49 @@ let undefined_location loc =
                       Const_base(Const_int line);
                       Const_base(Const_int char)]))
 
-let init_shape modl =
-  let rec init_shape_mod env mty =
+exception Initialization_failure of unsafe_info
+type init_result =
+  | Ok of lambda * lambda
+  | Fail of unsafe_info
+(* The Error constructor has been stolen earlier, we cannot use result *)
```

Comment:
You can use `Result.Error` to resolve the ambiguity, right? I'm not against defining new types if we think it is best/clearest, but if you really want `result` I think you should use it.

Reply:
I went ahead and used `result` finally.